### PR TITLE
[MGMT-1982] Proxy value must contain a scheme

### PR DESCRIPTION
**Pull request by masayag**
Signed-off-by: Moti Asayag <masayag@redhat.com>

### CODE REVIEW

**Comment by avishayt** [comment]
Looks OK to me, Vitaly please take a look

File: internal/cluster/validations/validations.go [hunk 1]
```diff
@@ -175,13 +176,20 @@ func ValidateClusterNameFormat(name string) error {
...
 func ValidateHTTPProxyFormat(proxyURL string) error {
-	if strings.HasPrefix(proxyURL, "https") {
+	if !govalidator.IsURL(proxyURL) {
+		return fmt.Errorf("Proxy URL format is not valid: '%s'", proxyURL)
+	}
+	u, err := url.Parse(proxyURL)
+	if err != nil {
+		return fmt.Errorf("Proxy URL format is not valid: '%s'", proxyURL)
+	}
+	if u.Scheme == "https" {
```
Comment (eranco74) [inline internal/cluster/validations/validations.go:186]: Don't think this is required due to line 189

**Review by eranco74 — COMMENTED** [review]
/lgtm

File: internal/cluster/validations/validations.go (hunk 1 above)
Comment (masayag) [reply internal/cluster/validations/validations.go:186]: no, according to the documentation [1], the description of the httpsProxy is:

> A proxy URL to use for creating HTTPS connections outside the cluster. If this is not specified, then httpProxy is used for both HTTP and HTTPS connections. The URL scheme must be http; https is currently not supported.

I would like to have a specific message for this case.

[1] https://docs.openshift.com/container-platform/4.5/networking/enable-cluster-wide-proxy.html#nw-proxy-configure-object_config-cluster-wide-proxy

**Pull request merged**